infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/DataSender.java

```diff
@@ -30,6 +33,12 @@
      */
     CompletableFuture<LogReplicationEntry> send(List<LogReplicationEntry> messages);
 
+    /**
+     * Used by Snapshot Full Sync to poll the receiver's status.
+     * @return
+     */
+    public LogReplicationQueryMetadataResponse sendQueryMetadataRequest() throws ExecutionException, InterruptedException;
```

Comment:
Why we don't use CompletableFuture<LogReplicationQueryMetadataResponse> as the return type?
The public accessor is redundant in an interface.